File: Week_01/G20200343030556/26. 删除排序数组中的重复项.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        # 法一：利用组成字典来去重
        # 内存说明：nums[:]使用的还是nums原来的内存，没有开辟新的内存空间
        # nums[:] = dict.fromkeys(nums).keys()
        # return len(nums)

        # 法二：快慢指针
        if nums is None: return 0
        size = len(nums)
        i = 0
        for j in range(1, size):
            if nums[j] != nums[i]:
                i, nums[i] = i + 1, nums[j]
        return len(nums)

File: Week_01/G20200447010070/26.删除排序数组中的重复项.js
/*
 * @lc app=leetcode.cn id=26 lang=javascript
 *
 * [26] 删除排序数组中的重复项
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  if (nums.length === 0) return 0;

  let i = 0;
  for (let j = 1; j < nums.length; j++) {
    if (nums[i] != nums[j]) {
      i++;
      nums[i] = nums[j];
    }
  }
  return i + 1;
};
// @lc code=end
